src/silx/io/nxdata/_utils.py

```diff
@@ -201,3 +199,17 @@ def validate_number_of_axes(
                 + "must define %d or %d axes." % (ndims, INTERPDIM[interpretation])
             )
     return issues
+
+
+def get_dataset_name(nxdata_group: h5py.Group, dataset_name: str | None) -> str | None:
```

Comment:
`get_dataset_name` function with a `dataset_name` argument sounds a bit unclear to me at first.
Maybe use a more explicit function name, `get_dataset_long_name`, `get_dataset_display_name`, ...? 🤷 

Also, sounds clearer to have:
`def get_dataset_name(dataset: h5py.Dataset) -> str:`
but it's at the expense of checking dataset_name is `None` outside of this function.. so not sure it worth it

Reply:
I'll sleep on it and try to find a better name for the function or the argument

Reply:
We use the term `label` in H5Web: https://github.com/silx-kit/h5web/blob/8c0f462223c5b0a0df21d1a02c99dc619a68df54/packages/app/src/vis-packs/nexus/utils.ts#L279

Reply:
Indeed, this is what is used for in the end.

Reply:
Well, if we go for `label`, ideally, it would have to be consistent. This means changing the names of the attributes `signal_name`, `axes_names` of the `NXdata` class

Reply:
I'm more arguing that you need both a base "name" (the name of the dataset in the HDF5 file) and a display "label" (long name with units), as you may need the refer back to the base name in some places (tooltip, etc.) Same with axes.

Of course, this does have broader implications, as the term `label` seems to already be used in a few places (e.g. `_NXdataStackView.setData()`, `ArrayStackPlot`, etc.) But I don't necessarily think this normalisation is critical for the work at hand.